packages/db/src/core/update.ts

```diff
@@ -14,39 +14,47 @@ import { Table } from "./createTable";
  * // Increase the age of Alice
  * await update(userTable, { id: userId, age: 16 });
  */
-export async function update<T, P extends keyof T>(
+export async function update<T extends object, P extends keyof T>(
   table: Table<T, P>,
   diff: Diff<T, P>
 ): Promise<void> {
   const primaryKeyProperty = table[BlinkKey].options.primary;
-  const primaryKey = diff[primaryKeyProperty] as unknown as T[P];
+  const primaryKey = diff[primaryKeyProperty];
+
+  if (primaryKey === undefined || primaryKey === null) {
+    throw new Error(`"${primaryKey}" is an invalid primary key value.`);
+  }
+
   const item = table[BlinkKey].storage.primary.get(primaryKey);
 
   if (item === undefined || item === null) {
     throw new Error(`Item with primary key "${primaryKey}" not found.`);
   }
 
   const oldItem = clone(item);
-  for (let key in diff) {
-    if (key === primaryKeyProperty) {
+
+  let key: keyof T;
+  for (key in diff) {
+    const diffValue = diff[key];
+    if (key === primaryKeyProperty || diffValue === undefined || diffValue === null) {
```

Comment:
This is questionable however. What is the expected behaviour when a value in diff is null or undefined?

Reply:
Expected is that the property value of the `item` should then be updated with null/undefined.
```ts
interface User {
  id: string;
  name: string;
  age?: number;
}

const aliceId = await insert(userTable, { id: uuid(), name: 'Alice', age: 30 });
await update(userTable, { id: aliceId, age: undefined });

const alice = await one(userTable, { id: aliceId });
expect(alice).toStrictEqual({ id: aliceId, name: 'Alice', age: undefined });
```

Reply:
For that to work the user will have to have `exactOptionalPropertyTypes` enabled in their tsconfig (otherwise [this happens](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=false#code/C4TwDgpgBAKlC8UDeAoKUCWATAXFAdgK4C2ARhAE4DcaUAZgPYN4DOwFG+A5jeqQIYU8RMpRoBfFCggAPMAwrAooSFAAiGOnQA8MADRQAClFnAI+LCygBrCCAZ1YAPgRHBwDPwA22gPLEMYF0DQycXADIoACUIAEdCDAoILG1DDABja2CjMJoUdIZ8NmU8OERUdGw8AEY9WkZmKAByBqa6vkE8AGYAOmqAFjrxPIB6EagGa3zC4qxNOmq8DS1spuwml3LaKqha+qY8FqY2lEkUMYnrA1JCJQBlAAlfAFUAGTUocih+fBMKCgUBgARA0gZgrPwoGwONwDPgGEpoZwuFAAD5QQgWCB0TjJaZFJRzLQAJiW81W602yG2uF27XoBwxWJx+GSQyAA)), but yes, gotcha.

I've updated the pull request to allow for undefined and null when T allows for it. Took some time to do it type-safely though, wasn't easy.